monkey/monkey_island/cc/environment/environment_config.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import List, Dict
+
+from monkey_island.cc.resources.auth.auth_user import User
+from monkey_island.cc.consts import MONKEY_ISLAND_ABS_PATH
+from monkey_island.cc.environment.user_creds import UserCreds
+from monkey_island.cc.resources.auth.user_store import UserStore
+
+
+class EnvironmentConfig:
+    def __init__(self,
+                 server_config: str,
+                 deployment: str,
+                 user_creds: UserCreds,
+                 aws=None):
+        self.server_config = server_config
+        self.deployment = deployment
+        self.user_creds = user_creds
+        self.aws = aws
+
+    @staticmethod
+    def get_from_json(config_json: str) -> EnvironmentConfig:
+        data = json.loads(config_json)
+        return EnvironmentConfig.get_from_dict(data)
+
+    @staticmethod
+    def get_from_dict(dict_data: Dict) -> EnvironmentConfig:
+        user_creds = UserCreds.get_from_dict(dict_data)
+        aws = dict_data['aws'] if 'aws' in dict_data else None
+        return EnvironmentConfig(server_config=dict_data['server_config'],
+                                 deployment=dict_data['deployment'],
+                                 user_creds=user_creds,
+                                 aws=aws)
+
+    def save_to_file(self):
+        file_path = EnvironmentConfig.get_config_file_path()
+        with open(file_path, 'w') as f:
+            f.write(json.dumps(self.to_dict(), indent=2))
+
+    @staticmethod
+    def get_from_file() -> EnvironmentConfig:
+        file_path = EnvironmentConfig.get_config_file_path()
+        with open(file_path, 'r') as f:
+            config_content = f.read()
+        return EnvironmentConfig.get_from_json(config_content)
+
+    @staticmethod
+    def get_config_file_path() -> str:
+        return os.path.join(MONKEY_ISLAND_ABS_PATH, 'cc/server_config.json')
```

Comment:
This is a bug, should be `os.path.join(MONKEY_ISLAND_ABS_PATH, 'cc', 'server_config.json')`...

Or even better change MONKEY_ISLAND_ABS_PATH from a string to a Path (from pathlib) and then use 
```python
MONKEY_ISLAND_ABS_PATH.joinpath("cc").joinpath("server_config.json")
```

Reply:
Not a bug, but I agree.